### DISCUSSION TITLE
please help. error when compiling src

### DISCUSSION THREAD

**Discussion by myie9** [mozilla-mobile/firefox-ios#7706] [Q&A]
use release 29.2
Carthage update --platform iOS
*** Fetching swift-protobuf
2020-12-08 17:07:24.797 Carthage[8794:2945945] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Failed to set posix_spawn_file_actions for fd -1 at index 0 with errno 9'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff204936af __exceptionPreprocess + 242
	1   libobjc.A.dylib                     0x00007fff201cb3c9 objc_exception_throw + 48
	2   Foundation                          0x00007fff212239ee -[NSConcreteTask launchWithDictionary:error:] + 5269
	3   ReactiveTask                        0x0000000104b3333e _TFFFF12ReactiveTask10launchTaskFTVS_4Task13standardInputGSqGV13ReactiveCocoa14SignalProducerCSo6NSDataO6Result7NoError___GS2_GOS_9TaskEventS3__OS_9TaskError_U_FTGVS1_8ObserverGS6_S3__S7__CS1_19CompositeDisposable_T_U0_FTCS_P33_8622FA0C3FE9071537A42FE3BB905E344PipeS10__GS2_GS6_S3__S7__U_FTGS8_GS6_S3__S7__S9__T_ + 1678

**Comment by dnarcese** [comment]
Which Xcode version are you on?

**Reply by myie9** [reply]
Version 12.2 (12B45b). That's the newest version.  Dear Daniela.

**Reply by dnarcese** [reply]
We don't yet support 12.2. Give it a try on 12.1

**Reply by myie9** [reply]
thank Daniela . i'm downloading 12.1 xcode.

**Reply by myie9** [reply]
> We don't yet support 12.2. Give it a try on 12.1

Now my xcode Version is 12.1 (12A7403).

has this error:

./bootstrap.sh 

Xcode SDK: 14
XCode 12 version detected! ••• Please ensure this is correct. •••
*** Cloning onepassword-app-extension
*** Cloning A-Star
*** Cloning SDWebImage
*** Cloning SnapKit
*** Cloning swift-protobuf
*** Cloning SwiftKeychainWrapper
*** Cloning telemetry-ios
Parse error: unexpected trailing characters in line: github "Dev1an/A-Star"    ~> 3.0.0-beta-1
audited 617 packages in 3.067s
found 9 vulnerabilities (8 low, 1 high)
  run `npm audit fix` to fix them, or `npm audit` for details

**Reply by myie9** [reply]
> We don't yet support 12.2. Give it a try on 12.1

still has error on 12.1.

would u please upload your Carthage folder to a github repo for me to download?

*** Fetching SnapKit
*** Fetching sentry-cocoa


2020-12-10 21:12:30.669 carthage[83580:1328082] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Failed to set posix_spawn_file_actions for fd -1 at index 0 with errno 9'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff204826af __exceptionPreprocess + 242
	1   libobjc.A.dylib                     0x00007fff201ba3c9 objc_exception_throw + 48

**Comment by myie9** [comment]
These is another problem: 
carthage build iphonesimulator, but no Release-iphoneos generated.

FengdeMacBook-Pro-3:firefox-ios fengguo$ ./carthage_command.sh 
Xcode SDK: 14
XCode 12 version detected! ••• Please ensure this is correct. •••
*** Skipped downloading SwiftyJSON.framework binary due to the error:
	"Bad credentials. The API can't be accessed using username/password authentication. Please create a personal access token to access this endpoint: http://github.com/settings/tokens"
*** Checking out SwiftyJSON at "5.0.0"
*** xcodebuild output can be found in /var/folders/g0/fg976j3s577_9th4qqllw8lc0000gn/T/carthage-xcodebuild.i9kZYW.log
*** Building scheme "SwiftyJSON iOS" in SwiftyJSON.xcworkspace
Failed to write to /Volumes/MACDATA/gf/firefox-ios/Carthage/Build/iOS/SwiftyJSON.framework: Error Domain=NSCocoaErrorDomain Code=260 "The file “SwiftyJSON.framework” couldn’t be opened because there is no such file." UserInfo={NSFilePath=/Users/fengguo/Library/Developer/Xcode/DerivedData/SwiftyJSON-danajwtpmtzizjaxqtcdnqgaipuy/Build/Products/Release-iphoneos/SwiftyJSON.framework, NSUnderlyingError=0x7ff3d2608e70 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
FengdeMacBook-Pro-3:firefox-ios fengguo$ ls /Users/fengguo/Library/Developer/Xcode/DerivedData/SwiftyJSON-danajwtpmtzizjaxqtcdnqgaipuy/Build/Products/
Release-iphonesimulator

**Reply by dnarcese** [reply]
You are on the latest commit on main, right?

**Reply by myie9** [reply]
yes.
would u please upload your Carthage folder to a github repo for me to download?

**Reply by dnarcese** [reply]
I tried, but it is too large for a repo.

**Reply by isabelrios** [reply]
Sorry to jump it and if you have already tried that, but I had same issue after trying xcode 12.2, my issues was that I forgot to select the proper one again from command line: 
`sudo xcode-select --switch /Applications/Xcode12.1.app/Contents/Developer` with the correct path to your xcode...
Just in case that helps

**Reply by myie9** [reply]
> I tried, but it is too large for a repo.

how about just upload Carthage/Build/iOS/ for a repo

**Comment by myie9** [comment]
Thank u. It's ok to upgrade carthage version
